src/server/verification/verificationAPI.js

```diff
@@ -353,50 +355,132 @@ const setup = (app: Router, verifier: VerificationAPI, storage: StorageAPI) => {
   )
 
   /**
-   * @api {get} /verify/w3/logintoken get W3 login token for current user
-   * @apiName Get W3 Login Token
+   * @api {get} /verify/bonuses check if there is available bonuses to charge on user's wallet and do it
+   * @apiName Web3 Charge Bonuses
    * @apiGroup Verification
    *
    * @apiSuccess {Number} ok
-   * @apiSuccess {String} loginToken
    * @ignore
    */
   app.get(
-    '/verify/w3/logintoken',
+    '/verify/bonuses',
     passport.authenticate('jwt', { session: false }),
     wrapAsync(async (req, res, next) => {
-      const { user, log } = req
-      const logger = log.child({ from: 'verificationAPI - login/token' })
+      const log = req.log.child({ from: 'verificationAPI - verify/bonuses' })
+
+      const { user: currentUser } = req
+      let wallet_token = currentUser.w3Token
 
-      let loginToken = user.loginToken
+      if (!wallet_token) {
+        const secureHash = md5(currentUser.email + conf.secure_key)
 
-      if (!loginToken) {
-        const secureHash = md5(user.email + conf.secure_key)
-        const web3Response = await fetch(`${conf.web3SiteUrl}/api/wl/user`, {
+        const w3UserRes = await fetch(`${conf.web3SiteUrl}/api/wl/user`, {
```

Comment:
Maybe transferred to middleware or to some helper?
with 
`     const w3UserRes = await fetch(`${conf.web3SiteUrl}/api/wl/user`, {`

Reply:
done